### PR TITLE
Add all available request options

### LINKED ISSUE #38: client is not fully configable
Hi,
firstoff: this was the most modern guzzleBundle i could find, so props for that :)

I need to use 
   
 ```
  cookies: true
   verify: false
```

in my Guzzle Client.
This is not possible at the moment since the the clients config is passed via bundle extension and these two options are not respected.

So wouldnt it make sense to just open the client config with ->useAttributeAsKey('name')?

Or how do i set other options to the client?

### PR DESCRIPTION
**Pull request by t-geindre**
__BC break__:
 * `default_headers` option becomes `headers` to stick to guzzle configuration,
 * `redirects` option no longer exists, `allow_redirects` accepts boolean options instead.

Fix #38

### CODE REVIEW

**Comment by gbouyge** [comment]
:+1:

**Comment by cmoysan** [comment]
👍

**Comment by Oliboy50** [comment]
I think it conflicts with an existing configuration key : `base_uri`
because we will be able to set the base_uri in 2 different ways... we have to know/tell which one will be used if we set 2 differents `base_uri`

File: README.md [hunk 1]
```diff
@@ -189,11 +189,14 @@ m6web_guzzlehttp:
                 default_ttl: 3600            # defautl ttl for cache entry in seconds
                 use_header_ttl: false        # use the cache-control header to set the ttl
                 cache_server_errors: true    # at false, no server errors will be cached
-                cache_client_errors: true    # at false, no client errors will be cached 
+                cache_client_errors: true    # at false, no client errors will be cached
             default_headers:                 # optionnal. Default request headers
                 User_Agent: "m6web/1.0"      # set header "User-Agent" with the value "m6web/1.0"
                 header\_name: "my value"     # set header "header_name" with value "my value"
-
+            request_defaults:                # default request options, all request option available
```
Comment (Oliboy50) [inline README.md]: s/option/options

**Comment by gbouyge** [comment]
:+1:

File: README.md [hunk 2]
```diff
@@ -169,30 +169,80 @@ For more information on how to setup the RedisBundle, refer to the README in the
...
-                User_Agent: "m6web/1.0"      # set header "User-Agent" with the value "m6web/1.0"
-                header\_name: "my value"     # set header "header_name" with value "my value"
+            base_uri: ""                           # Base uri to prepend on request uri
+            timeout: 5.0                           # request timeout
+            http_errors: true                      # enable or disable exception on http errors
+            allow_redirects: true                  # enable or disable follow redirect
+            redirect_handler: curl                 # guzzle or curl
+            proxy: proxy:port                      # Optional. Set the proxy for client.
+            guzzlehttp_cache:                      # optional cache
+                service: my_cache_service          # reference to service who implements the cache interface
```
Comment (Oliboy50) [inline README.md]: cette référence devrait commencer par `@` du coup ?

File: README.md [hunk 3]
```diff
@@ -169,30 +169,80 @@ For more information on how to setup the RedisBundle, refer to the README in the
...
-                use_header_ttl: false        # use the cache-control header to set the ttl
-                cache_server_errors: true    # at false, no server errors will be cached
-                cache_client_errors: true    # at false, no client errors will be cached 
-            default_headers:                 # optionnal. Default request headers
-                User_Agent: "m6web/1.0"      # set header "User-Agent" with the value "m6web/1.0"
-                header\_name: "my value"     # set header "header_name" with value "my value"
+            base_uri: ""                           # Base uri to prepend on request uri
+            timeout: 5.0                           # request timeout
+            http_errors: true                      # enable or disable exception on http errors
+            allow_redirects: true                  # enable or disable follow redirect
```
Comment (Oliboy50) [inline README.md]: conflit avec le `allow_redirects` plus bas, non ?

**Comment by Oliboy50** [comment]
👍 à part mes 2 remarques sur la doc

**Comment by t-geindre** [comment]
v2.0.0

**Pull request merged**